### PR TITLE
rust methods that return multiple values

### PR DESCRIPTION
**Pull request by saolsen**

### CODE REVIEW

File: languages/rust/oso/src/host/to_polar.rs [hunk 1]
```diff
@@ -172,3 +172,19 @@ impl<C: ToPolarIter> ToPolarIter for Option<C> {
         )
     }
 }
+
+pub struct PolarValues<I, Iter>
+where
+    I: ToPolarIter + 'static,
```
Comment (dhatch) [inline languages/rust/oso/src/host/to_polar.rs:178]: Fun stuff. Should we call `ToPolarIter` something else?

Could revisit this separately.  If we never use it for more than one result, it seems like a trait along the lines of `to_polar_result() -> oso::Result<Option<Value>>` would be all we need?

File: languages/rust/oso/src/host/class.rs [hunk 2]
```diff
@@ -175,6 +175,20 @@ where
         self
     }
 
+    pub fn add_values_method<F, Args, I>(mut self, name: &str, f: F) -> Self
```
Comment (dhatch) [inline languages/rust/oso/src/host/class.rs:178]: `add_iterator_method`? With a docstring indicating the generation of choices behavior.

**Review by dhatch — APPROVED** [review]
Some naming questions. We can revisit implementation when we improve memory model.

File: languages/rust/oso/src/host/to_polar.rs (hunk 1 above)
Comment (saolsen) [reply languages/rust/oso/src/host/to_polar.rs]: It is used for more than one thing now with the iterator stuff. I changed the names to hopefully be better.

File: languages/rust/oso/src/host/class.rs (hunk 2 above)
Comment (saolsen) [reply languages/rust/oso/src/host/class.rs]: :+1:

**Pull request merged**